File: go/2580.go
package main

import "slices"

func countWays(ranges [][]int) int {
	const MOD = 1_000_000_007
	slices.SortFunc(ranges, func(a, b []int) int { return a[0] - b[0] })
	ans, max_r := 1, -1
	for _, p := range ranges {
		l, r := p[0], p[1]
		if l > max_r {
			ans = (ans * 2) % MOD
		}
		max_r = max(max_r, r)
	}
	return ans
}

File: python/2580.py
from typing import List

class Solution:
    def countWays(self, ranges: List[List[int]]) -> int:
        MOD = 1_000_000_007
        ranges.sort(key=lambda x: x[0])
        ans = 1
        max_r = -1
        for l, r in ranges:
            if l > max_r:
                ans = (ans * 2) % MOD
            max_r = max(max_r, r)
        return ans